src/platform/datapath_kqueue.c

```diff
@@ -1127,6 +1127,7 @@ CxPlatSocketContextRecvComplete(
 
     RecvPacket->Route->Queue = SocketContext;
     RecvPacket->TypeOfService = 0;
+    RecvPacket->HopLimitTTL = 0; // TODO: Check to see if it's even possible to get this information on macOS and FreeBSD. If so, implement it. Otherwise, document that it's not supported.
```

Comment:
https://man.freebsd.org/cgi/man.cgi?query=ip&sektion=4&manpath=FreeBSD+9.0-RELEASE

See "IP_RECVTTL" in this page. I think it's supported.

Reply:
Added a feature flag for IP_TTL support or not depending on the platform. Decided to hold off on MacOs enablement.